### DISCUSSION TITLE
Can we try to unify handling of *_proxy environment variables across different tools and languages?

### DISCUSSION THREAD

**Discussion by piotr-dobrogost** [curl/curl#11962] [General]
Handling of `*_proxy` environment variables is not well documented (https://superuser.com/q/944958/664, https://github.com/curl/curl/issues/1208, https://github.com/curl/curl/issues/2773) and varies across various tools and languages (https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/ by @stanhu based on research in https://gitlab.com/gitlab-com/support/support-team-meta/-/issues/2991).

Would it be possible to coordinate with maintainers of other tools and languages mentioned above (wget, Ruby, Python, Go) and unify handling of *_proxy environment variables?

ps.
Could someone please link to this discussion from both https://github.com/curl/curl/issues/1208 and https://github.com/curl/curl/issues/2773 as they are locked and mentioning them here will not result in creation of links pointing from them back here?

**Comment by bagder** [comment]
> Would it be possible to coordinate with maintainers of other tools and languages mentioned above (wget, Ruby, Python, Go) and unify handling of *_proxy environment variables?

I cannot speak for anyone else than myself. I would not mind trying to improve the situation and would participate in such a "coordination". But...

1. I bet it will be hard since everyone is likely to be against breaking behavior for their own users. I am.
2. I have no idea how keen others are on working on this
3. The above linked materials are already outdated, at least in terms of what curl does/supports

**Reply by piotr-dobrogost** [reply]
ad. 1 Don't you think that cleaning up this _mess_ for current and **all future users** is worth breaking behavior (especially taking into consideration that adjusting to the new behavior for the user would be as easy as changing casing and/or adding/removing one or two environment variables? There is no hurry, the change in behavior can be announced with much advance.
ad. 2 Sure thing, neither do I but I had to start somewhere :)
ad. 3 Could you point to documentation of current curl behavior with regard to `*_proxy` variables?

In addition could you please link to this discussion from both https://github.com/curl/curl/issues/1208 and https://github.com/curl/curl/issues/2773 as they are locked and mentioning them here will not result in creation of links pointing from them back here? (I'm not sure but unlocking and locking again those issues could do the job).

**Reply by bagder** [reply]
> Don't you think that cleaning up this mess for current and all future users is worth breaking behavior

It would depend exactly on what change we talk about. But since we all probably think like that, there's a big risk there are things we will not be able to agree to.

> Could you point to documentation of current curl behavior with regard to *_proxy variables

The curl.1 man page (towards the end)

> could you please link to this discussion

Linking from old  and closed bug reports have extremely little value. Just forget those.
